src/Tasks.UnitTests/Copy_Tests.cs

```diff
@@ -515,6 +515,51 @@ public void DoCopyOverDifferentFile()
             }
         }
 
+        /*
+         * Method:   DoCopyOverCopiedFile
+         *
+         * If SkipUnchangedFiles is set to "false" then we should still be able to copy over files (or links)
+         * that have same dates and sizes.
+         */
+        [Fact]
+        public void DoCopyOverCopiedFile()
```

Comment:
I think it's worth adding an additional test locking in the behaviour that symlinks get deleted and recreated even when `SkipUnchangedFiles = true`.

Reply:
Done, but I've found that `SkipUnchangedFiles` doesn't work on Linux.
On Linux file attributes are not preserved when file is copied, so `IsMatchingSizeAndTimeStamp` will always return false.
I'm happy to fix it, but I don't think it should be done in this PR.

Reply:
Agreed. Can you please open a separate issue for Linux?
Thank you for the investigation!

Reply:
Done, : https://github.com/Microsoft/msbuild/issues/4015